File: Python/AcmQuiz001.py
def quiz3Nplus1(n,mSum):
    if n <= 1:
        return mSum
    elif n%2==1:
        return quiz3Nplus1(n*3+1,mSum+1)
    else:
        return quiz3Nplus1(n/2,mSum+1)

if __name__ =='__main__':
    i = 1
    j = 10
    mMax = 0
    for q in range(i,j):
        tmp = quiz3Nplus1(q,1)
        if tmp > mMax:
            mMax = tmp
    print "max len : "+str(mMax)

File: Go/AcmQuiz001.go
package main

import "fmt"

func quiz3Nplus1(n int, mSum int) int {
	if n <= 1 {
		return mSum
	} else if n%2 == 1 {
		return quiz3Nplus1(n*3+1, mSum+1)
	} else {
		return quiz3Nplus1(n/2, mSum+1)
	}
}

func main() {
	max := 0
	nums := []int{1, 2, 3, 4, 5, 6, 7, 8, 9}
	for q := range nums {
		tmp := quiz3Nplus1(q, 1)
		if tmp > max {
			max = tmp
		}
	}
	fmt.Printf("max len : %d\n", max)
}
